### PR TITLE
Fix ValueAtPoints data type

### PR DESCRIPTION
**Pull request by tjhei**
max_values_in_array should be an integer not a double.

Also update schema files.

### CODE REVIEW

**Review by gassmoeller — APPROVED** [review]

**Comment by tjhei** [comment]
FYI: @danieldouglas92

**Comment by tjhei** [comment]
Hm, doesn't compile on the tester?!

**Comment by MFraters** [comment]
hmm, so it turns out that rapidjson doesn't support size_t for this part of the conversion directly:

https://github.com/GeodynamicWorldBuilder/WorldBuilder/blob/d9c2f64e37b42aa3046ccd9b093830938afaaec9/include/rapidjson/document.h#L987-L1050

We could use a unsigned int or int64_t instead, or cast it to a int64_t where  it is used, or try to implement it.

**Comment by MFraters** [comment]
@tjhei are you oke with this solution, turining the `size_t` into `uint64_t`?

**Comment by tjhei** [comment]
Yes, sorry didn't get time to update yet

**Comment by MFraters** [comment]
I updated it for you :)

File: source/world_builder/features/subducting_plate_models/temperature/mass_conserving.cc [hunk 1]
```diff
@@ -117,7 +117,7 @@ namespace WorldBuilder
           prm.declare_entry("density", Types::Double(3300),
                             "The reference density of the subducting plate in $kg/m^3$");
 
-          prm.declare_entry("plate velocity", Types::OneOf(Types::Double(0.01),Types::Array(Types::ValueAtPoints(0.01, std::numeric_limits<size_t>::max()))),
+          prm.declare_entry("plate velocity", Types::OneOf(Types::Double(0.01),Types::Array(Types::ValueAtPoints(0.01, std::numeric_limits<uint64_t>::max()))),
```
Comment (tjhei) [inline source/world_builder/features/subducting_plate_models/temperature/mass_conserving.cc:120]: Why do you use uint max for doubles?

Comment (MFraters) [reply source/world_builder/features/subducting_plate_models/temperature/mass_conserving.cc:120]: That contains the max number of elements: https://github.com/GeodynamicWorldBuilder/WorldBuilder/blob/d9c2f64e37b42aa3046ccd9b093830938afaaec9/source/world_builder/types/value_at_points.cc#L31-L33

**Pull request merged**